akka-stream-tests/src/test/scala/akka/stream/FusingSpec.scala

```diff
@@ -84,6 +90,71 @@ class FusingSpec extends StreamSpec {
       refs.toSet should have size (in.size + 1) // outer/main actor + 1 actor per subflow
     }
 
+    //an UnfoldResourceSource equivalent without an async boundary
+    case class UnfoldResourceNoAsyncBoundry[T, S](create: () => S, readData: (S) => Option[T], close: (S) => Unit)
+        extends GraphStage[SourceShape[T]] {
+      val stage_ = new UnfoldResourceSource(create, readData, close)
+      override def initialAttributes: Attributes = Attributes.none
+      override val shape = stage_.shape
+      def createLogic(inheritedAttributes: Attributes) = stage_.createLogic(inheritedAttributes)
+      def asSource = Source.fromGraph(this)
+    }
+
+    "propagate downstream errors through async boundary" in {
+      val promise = Promise[Done]
+      val slowInitSrc = UnfoldResourceNoAsyncBoundry(
+        () => { Await.result(promise.future, duration.Duration.Inf); () },
```

Comment:
I'd prefer some finite timeout, in case the test is not working as expected. such as `1.minute`